worker/src/vectorize.rs

```diff
@@ -0,0 +1,319 @@
+    /// Return vectors from the specified namespace. Default `none`.
+    namespace: Option<String>,
+    /// Return vector values. Default `false`.
+    return_values: bool,
+    /// Return vector metadata. Default `None`.
+    return_metadata: VectorizeMetadataRetrievalLevel,
+    /// Default `none`.
+    filter: Option<VectorizeVectorMetadataFilter>,
+}
+
+impl VectorizeQueryOptions {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn with_top_k(mut self, top_k: u8) -> Self {
+        self.top_k = top_k;
+        self
+    }
+
+    pub fn with_namespace(mut self, namespace: &str) -> Self {
+        self.namespace = Some(namespace.to_owned());
+        self
+    }
+
+    pub fn with_return_values(mut self, return_values: bool) -> Self {
+        self.return_values = return_values;
+        self
+    }
+
+    pub fn with_return_metadata(
+        mut self,
+        return_metadata: VectorizeMetadataRetrievalLevel,
+    ) -> Self {
+        self.return_metadata = return_metadata;
+        self
+    }
+
+    pub fn with_filter_entry<T: Serialize>(
+        mut self,
+        key: &str,
+        op: VectorizeVectorMetadataFilterOp,
+        value: T,
+    ) -> Result<Self> {
+        let mut filter = self.filter.unwrap_or_default();
+        let inner = filter.entry(key.to_owned()).or_default();
+        inner.insert(op, serde_json::to_value(value)?);
+        self.filter = Some(filter);
+        Ok(self)
+    }
+}
+
+impl Default for VectorizeQueryOptions {
+    fn default() -> Self {
+        Self {
+            top_k: 5,
+            namespace: None,
+            return_values: false,
+            return_metadata: VectorizeMetadataRetrievalLevel::None,
+            filter: None,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+/// Represents a single vector value set along with its associated metadata.
+pub struct VectorizeVectorResult {
+    /// The ID for the vector. This can be user-defined, and must be unique. It should uniquely identify the object, and is best set based on the ID of what the vector represents.
+    pub id: String,
+    /// The vector values.
+    pub values: Option<Vec<f32>>,
+    /// Metadata associated with the vector. Includes the values of other fields and potentially additional details.
+    pub metadata: Option<serde_json::Map<String, serde_json::Value>>,
+    /** The namespace the vector belongs to. */
+    pub namespace: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct VectorizeMatchVector {
+    #[serde(flatten)]
+    pub vector: VectorizeVectorResult,
+    /// The score or rank for similarity, when returned as a result
+    pub score: Option<f64>,
+}
+
+#[derive(Debug, Deserialize)]
+/// A set of matching {@link VectorizeMatch} for a particular query.
+pub struct VectorizeMatches {
+    pub matches: Vec<VectorizeMatchVector>,
+    pub count: u64,
+}
+
+/// A Vectorize Vector Search Index for querying vectors/embeddings.
+pub struct VectorizeIndex(VectorizeIndexSys);
```

Comment:
The workers-types package seems to suggest that Vectorize v2 is just called `Vectorize` for the type name with `VectorizeIndex` being the old binding, i think we need to change the name here

<img width="870" alt="image" src="https://github.com/user-attachments/assets/eb442086-ff51-4ec2-8fb7-a1ee707c5b7a">

Reply:
Nice catch! Fixed.